kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -25,16 +25,17 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(KaldbDistributedQueryService.class);
 
-  public static List<String> servers = new ArrayList<>();
+  private final SearchMetadataStore searchMetadataStore;
 
-  private final LoadingCache<String, KaldbServiceGrpc.KaldbServiceFutureStub> stubLoadingCache =
-      CacheBuilder.newBuilder().build(CacheLoader.from(this::getKaldbServiceGrpcClient));
+  // Number of times the listener is fired
+  public static final String SEARCH_METADATA_TOTAL_CHANGE_COUNTER =
+      "search_metadata_total_change_counter";
+  private final MeterRegistry meterRegistry;
+  private final Counter searchMetadataTotalChangeCounter;
 
-  private KaldbServiceGrpc.KaldbServiceFutureStub getKaldbServiceGrpcClient(String server) {
-    return Clients.newClient(server, KaldbServiceGrpc.KaldbServiceFutureStub.class)
-        .withCompression("gzip")
-        .withInterceptors(GrpcTracing.newBuilder(Tracing.current()).build().newClientInterceptor());
-  }
+  private Map<String, KaldbServiceGrpc.KaldbServiceFutureStub> stubs = new ConcurrentHashMap<>();
+
+  private static String protocol = "gproto+http://";
```

Comment:
Please move this to toURL into the `SeachContext`.

Reply:
done